### ISSUE TITLE
Codecov: utils-0.57/lib/combinat.gi vs lib/combinat.gi

### ISSUE THREAD

**Issue by fingolfin** [gap-system/gap#2743]
The coverage data reported on 
https://codecov.io/gh/gap-system/gap/src/master/lib/combinat.g clearly is nonsense.i

I strpngly suspect that here data for utils-0.57/lib/combinat.gi is being mixed up with the actual relevant data.

**Comment by cdwensley** [comment]
The Utils file can easily be renamed if that would help.

**Comment by fingolfin** [comment]
No, I don't think modifying the `utils` package here is the way to go; while that would solve this particular instance of the problem, it'd still cure the symptom, not the cause of the issue.

One can download the logs of an affected build, e.g. from here: <https://codecov.io/gh/gap-system/gap/commit/f701a6f1671e5133792a9f026be3cd0ff7b4061e/build>. The string `combinat.gi` occurs three times:
```
...
lib/combinat.gd
lib/combinat.gi
...
"/home/travis/build/gap-system/gap/lib/combinat.gi": {
"27": "0",
...
"/home/travis/build/gap-system/gap/pkg/utils-0.57/lib/combinat.gi": {
"18": "0",
```
So the paths are reported correctly to codecov. But the heuristic employed by codecov to match paths clearly fails to recognize these as separate files (note that it *has* to try to ignore part of the path, as it is meant to integrate coverage results from multiple sources, such as Travis and Codecov; in this example, it is "just" overeager).

I'll contact codecov support, perhaps they can help us.

**Comment by fingolfin** [comment]
So far I only got generic, unhelpful replies which made it sound to me as if they hadn't even read my request. I was told that they recently (weekend August 25/26) "went through our migration", but the "conversation" eventually dried up. A second recent request informing them that viewing PR data on codecov times out was eventually acknowledged with "we are working on it". But I am have not very high hopes: in the past, I learned that we are apparently a very big "client", and our data sets are simply too big for them, leading to time outs (which is a bit ridiculous, as there are *way* bigger projects out there; if they can't deal with ours, how will they scaler?).

**Comment by cdwensley** [comment]
While revising Utils I found it makes sense to move the code in combinat.g{d,i} to lists.g{d,i} in line with the tests and documentation - so these files will now disappear.

**Comment by fingolfin** [comment]
Note that the same problem affects `src/io.c` -- the io package also has a file called `io.c`

This really needs to be resolved on the Codecov side.
